safemode/safemode.go

```diff
@@ -0,0 +1,255 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package safemode
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	chaostypes "github.com/DataDog/chaos-controller/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Safemode interface {
+	// CreationSafetyNets will look at all the specific safety nets corresponding to the creation of a new disruption
+	CreationSafetyNets() ([]string, error)
```

Comment:
This is currently the only place where we are checking and enforcing safety for now, yes? Then I assume the name is to differentiate it from as-yet-unimplemented RuntimeSafetyNets that are checked in an ongoing fashion? 🚲  🏠 , but I'd prefer something like `CheckSpecSafety`, as this method appears to be a verb doing the checking, rather than creating safety nets that will later be checked

Reply:
This is a fair comments, as the method doesnt create safety nets to be used later. Though the checks that are being done in `CreationSafetyNets` are ONLY checked during the start of the disruption rather than during the runtime of the disruption. Do you think `CheckInitialSafety` with its counterpart `CheckRuntimeSafety` would be more appropriate?